integration/daemon/default_storage_test.go

```diff
@@ -79,6 +79,8 @@ func TestGraphDriverPersistence(t *testing.T) {
 	assert.Check(t, is.Equal(imageInspect.GraphDriver.Name, prevDriver), "Image graphdriver data should match")
 
 	// Verify our container is still there
-	_, err = c.ContainerInspect(ctx, containerID)
+	containerInspect, err := c.ContainerInspect(ctx, containerID)
 	assert.NilError(t, err, "Test container should still exist after daemon restart")
+	assert.Check(t, containerInspect.GraphDriver != nil, "GraphDriver should be set for graphdriver backend")
+	assert.Check(t, is.Equal(containerInspect.GraphDriver.Name, prevDriver), "Container graphdriver data should match")
```

Comment:
For a follow-up; we should check if we have a similar test for snapshotters;


- start a daemon with snapshotter X
- stop the daemon
- start the daemon with snapshotter Y (e.g. "vis" or equivalent)
- verify that we didn't try to switch the containers / images to a different snapshotter

(well; not sure if we support this scenario, but at least the stop/start as this test does)